data-prepper-plugins/s3-source/src/main/java/com/amazon/dataprepper/plugins/source/SqsWorker.java

```diff
@@ -110,16 +114,35 @@ private List<S3EventNotification.S3EventNotificationRecord> convertS3EventMessag
     }
 
     private void processS3ObjectAndDeleteSqsMessages(final Map<Message, List<S3EventNotification.S3EventNotificationRecord>> s3EventNotificationRecords) {
+        List<DeleteMessageBatchRequestEntry> deleteMessageBatchRequestEntryCollection = new ArrayList<>();
         for (Map.Entry<Message, List<S3EventNotification.S3EventNotificationRecord>> entry: s3EventNotificationRecords.entrySet()) {
             if (entry.getValue().isEmpty()) {
-                // TODO: delete or ignore based on configuration
+                if (s3SourceConfig.getOnErrorOption().equals(OnErrorOption.DELETE_MESSAGES)) {
+                    deleteMessageBatchRequestEntryCollection.add(buildDeleteMessageBatchRequestEntry(entry.getKey()));
+                }
             }
             else if (isEventNameCreated(entry.getValue().get(0))) {
                 final S3ObjectReference s3ObjectReference = populateS3Reference(entry.getValue().get(0));
                 s3Service.addS3Object(s3ObjectReference);
-                // TODO: delete sqsMessages which are successfully processed
+                deleteMessageBatchRequestEntryCollection.add(buildDeleteMessageBatchRequestEntry(entry.getKey()));
             }
         }
+        DeleteMessageBatchRequest deleteMessageBatchRequest = buildDeleteMessageBatchRequest(deleteMessageBatchRequestEntryCollection);
+        sqsClient.deleteMessageBatch(deleteMessageBatchRequest);
```

Comment:
We should not make this request if `deleteMessageBatchRequestEntryCollection` is empty. A couple reasons:

1. I'm fairly sure the API will reject the request.
2. Even if the API accepts it, it will incur an AWS cost since SQS billing is based on number of requests.

Reply:
Yeah, good catch. I'll update it